src/server.c

```diff
@@ -4105,7 +4107,7 @@ int processCommand(client *c) {
         if (listLength(server.ready_keys))
             handleClientsBlockedOnKeys();
     }
-
+    c->db->current_dict = NULL;
```

Comment:
Is `c->db` always the same before and after executing the command? Do we need to consider temb DBs and do something there? (swapMainDbWithTempDb) Hopefully not but I'm not sure...

Reply:
I'm not aware of any scenario when it would be an issue, main processing loop is single threaded and next command is not starting until previous one completes. If you know any use case when this might be an issue, please let me know.

Reply:
You're probably right. There's no way for the client to change db either (SELECT not possible in cluster mode).

It just looks risky somehow. If the db changes, the old db may stick around somewhere with a current_dict set. Can we add some assert to check that db hasn't changed? E.g.

    if (c->slot >= 0) serverAssert(c->db->current_dict == c->db->dict[c->slot];

Possibly check that it's NULL before we set it too?

Not sure if this makes sense. Perhaps I'm paranoid.

Reply:
We can add that sanity check. I didn't like these `if (c->slot >= 0)` checks here, we can eliminate them by saving current slot instead of current dict. Then assert would simply need to compare two ints.

Reply:
I've added this check, and also to make it nicer, I've cached slot id, instead of dict, this way we can simply compare integers in the assert. I've found a couple of interesting things related to scripts though:
* It's possible that some keys were not declared by the script, in which case slot is set in the client during script execution, hence cached slot id will be -1, but actual slot will be different (see `scriptVerifyClusterState` function). To avoid this we can add a simple check before the assert to only perform assertion when the slot was cached.
* Scripts allow multi slot commands, when appropriate flags are set, which means that caching single slot id is inappropriate. I think best we can do is to identify this situation during script execution and clear cached slot id so that fallback logic that calculates key hash is used in the `getDict` function.